tests/paramgrill.c

```diff
@@ -632,131 +1391,1112 @@ int benchFiles(const char** fileNamesTable, int nbFiles)
+    if(memoTable[memind] >= INFEASIBLE_THRESHOLD) {
+        return INFEASIBLE_RESULT; //see feasibleBenchMemo for concerns
+    } else {
+        const size_t blockSize = g_blockSize ? g_blockSize : srcSize;
+        U32 const maxNbBlocks = (U32) ((srcSize + (blockSize-1)) / blockSize) + (U32)nbFiles;
+        const void ** const srcPtrs = (const void** const)malloc(maxNbBlocks * sizeof(void*));
+        size_t* const srcSizes = (size_t* const)malloc(maxNbBlocks * sizeof(size_t));
+        void ** const dstPtrs = (void** const)malloc(maxNbBlocks * sizeof(void*));
+        size_t* const dstCapacities = (size_t* const)malloc(maxNbBlocks * sizeof(size_t));
+        U32 nbBlocks;
+        int res;
+
+        if(!srcPtrs || !srcSizes || !dstPtrs || !dstCapacities) {
+            free(srcPtrs);
+            free(srcSizes);
+            free(dstPtrs);
+            free(dstCapacities);
+            DISPLAY("Allocation Error\n");
+            return ERROR_RESULT;
+        }
+
+        {   
+            const char* srcPtr = (const char*)srcBuffer;
+            char* dstPtr = (char*)dstBuffer;
+            size_t dstSizeRemaining = dstSize;
+            U32 fileNb;
+            for (nbBlocks=0, fileNb=0; fileNb<nbFiles; fileNb++) {
+                size_t remaining = fileSizes[fileNb];
+                U32 const nbBlocksforThisFile = (U32)((remaining + (blockSize-1)) / blockSize);
+                U32 const blockEnd = nbBlocks + nbBlocksforThisFile;
+                if(remaining > dstSizeRemaining) {
+                    DEBUGOUTPUT("Warning: dstSize too small to benchmark completely \n");
+                    remaining = dstSizeRemaining;
+                    dstSizeRemaining = 0;
+                } else {
+                    dstSizeRemaining -= remaining;
+                }
+                for ( ; nbBlocks<blockEnd; nbBlocks++) {
+                    size_t const thisBlockSize = MIN(remaining, blockSize);
+                    srcPtrs[nbBlocks] = (const void*)srcPtr;
+                    srcSizes[nbBlocks] = thisBlockSize;
+                    dstPtrs[nbBlocks] = (void*)dstPtr;
+                    dstCapacities[nbBlocks] = ZSTD_compressBound(thisBlockSize);
+                    srcPtr += thisBlockSize;
+                    dstPtr += dstCapacities[nbBlocks];
+                }
+                if(!dstSize) { break; }   
+            }   
+        }
+
+        res = infeasibleBench(resultPtr, srcPtrs, srcSizes, dstPtrs, dstCapacities, nbBlocks, dictBuffer, dictSize, ctx, dctx, 
+               cParams, target, winnerResult);
+        if(res == FEASIBLE_RESULT) {
+            memoTable[memind] = 255; 
+        }
+        free(srcPtrs);
+        free(srcSizes);
+        free(dstPtrs);
+        free(dstCapacities);
+        return res;
+    }
+}
+
+/* specifically feasibleBenchMemo and infeasibleBenchMemo */
+//maybe not necessary 
+typedef int (*BMK_benchMemo_t)(BMK_result_t*, const void*, size_t, void*, size_t, ZSTD_CCtx*, ZSTD_DCtx*, 
```

Comment:
while parameter names are not strictly necessary,
they help understanding what each parameter is about.